modules/hal_nordic/Kconfig

```diff
@@ -201,11 +180,27 @@ config NRF_802154_RX_BUFFERS
 	  Number of buffers in nRF 802.15.4 driver serialization host's receive queue.
 	  If this value is modified, its remote counterpart must be set to the exact same value.
 
-endif
+config NRF_802154_PENDING_SHORT_ADDRESSES
+	int "nRF 802.15.4 pending short addresses"
+	default 16
+	help
+	  Number of slots containing short addresses of nodes for which pending data is stored
 
-endmenu # NRF_802154_SER_HOST || NRF_802154_SER_RADIO
+config NRF_802154_PENDING_EXTENDED_ADDRESSES
+	int "nRF 802.15.4 pending extended addresses"
+	default 16
+	help
+	  Number of slots containing extended addresses of nodes for which pending data is stored
 
-if NRF_802154_RADIO_DRIVER || NRF_802154_SERIALIZATION
+config NRF_802154_ENCRYPTION
+	bool "nRF 802.15.4 AES-CCM* authentication & encryption"
+	depends on !CRYPTO_NRF_ECB
```

Comment:
What we want to do is to control if nRF 802.15.4 Radio Driver (regardless if looking on it through serialization or not) supports encryption. How does this `depends on !CRYPTO_NRF_ECB` correspond to both cases? `CRYPTO_NRF_ECB` can have one value on NET and another on APP, or am I missing sth?

Reply:
That's a good point, thank you. Looking at its definition, `CRYPTO_NRF_ECB` definitely can have different value on the network core than on the application core since only the former has ECB peripheral, which this symbol depends on. This could lead to a different setting of `NRF_802154_ENCRYPTION` on both cores: `y` on the application core where the `depends on` above is harmless and `n` on the network core, where the crypto driver could prevent `NRF_802154_ENCRYPTION` from being set. Such error could be missed easily as it would only generate a build warning.

That said, I don't know what's the best way to improve that solution. It seems to me that this issue stems from the build system's deficiency in representing relations between configurations of different applications for different targets, which in reality represent the same capabilities in a multicore architecture.

Let me know what's your opinion on potential fixes.

Reply:
Sysbuild was supposed to allow between-CPUs configuration dependencies. I'm not sure if it was implemented though.

Reply:
yes, that's build system limitation, let's go with it as it